pkg/apis/core/v1beta1/types_utils.go

```diff
@@ -70,3 +72,29 @@ const (
 	// that the observed generation in a status is outdated.
 	OutdatedStatusError = "OutdatedStatus"
 )
+
+// ShootItems provides helper functions with ShootLists
+type ShootItems ShootList
+
+// Union returns a set of Shoots that presents either in s or shootList
+func (s *ShootItems) Union(shootList *ShootList) []Shoot {
+	unionedShoots := make(map[string]Shoot)
+	for _, s := range s.Items {
+		unionedShoots[objectKey(s.Namespace, s.Name)] = s
+	}
+
+	for _, s := range shootList.Items {
+		unionedShoots[objectKey(s.Namespace, s.Name)] = s
+	}
+
+	shoots := make([]Shoot, 0, len(unionedShoots))
+	for _, v := range unionedShoots {
+		shoots = append(shoots, v)
+	}
+
+	return shoots
+}
+
+func objectKey(namesapce, name string) string {
+	return fmt.Sprintf("%s%c%s", namesapce, '/', name)
+}
```

Comment:
Isn't the `helper` package a better place for such things?
Also, API-wise I feel it's a little inconsistent that a `ShootItems` structure has a `Union` function that doesn't take another `ShootItems` but a `ShootList`.

Reply:
ShootItems could be type alias of []Shoot instead of new type, wdyt? That way we can 
`var shootItems ShootItems`
`shootItems = shootList.Items`